submodules/telemetry/__init__.py

```diff
@@ -1,166 +1,129 @@
+    Clear the telemetry buffers - clearing general_queue, the log, and error stacks.
+    :return: Nothing
     """
-    global telem_packet_buffer, packet_lock
+    global packet_lock, log_stack, err_stack, general_queue
     with packet_lock:
-        for packet in packets:
-            telem_packet_buffer.append(packet)
+        general_queue.clear()
+        log_stack.clear()
+        err_stack.clear()
 
 
-def send(ignoreADCS=False, radio='aprs'):
+def decide() -> None:
     """
-    Concatenates packets to fit in max_packet_size (defined in config) and send through the APRS, dequing the packets in the process
-    :param ignoreADCS: If true, ignores ADCS canTJBeSeen.
-    :param radio: Radio to send telemetry over, either "aprs" or "iridium"
+    A thread method to constantly check general_queue for messages and process them if there are any.
+    :return: Nothing
     """
-    global packetBuffers, event_packet_buffer, telem_packet_buffer, packet_lock
-    squishedpackets = ""
-
-    with packet_lock:
-        # packet_lock.acquire()
-        # TODO while len(event_packet_buffer) + len(telem_packet_buffer) > 0 and (adcs.can_TJ_be_seen() or ignoreADCS):
-        while len(event_packet_buffer) + len(telem_packet_buffer) > 0:
-            for buffer in packetBuffers:
-                # TODO while len(buffer) > 0 and len(squishedPackets) < config['telemetry']['max_packet_size'] and (adcs.can_TJ_be_seen() or ignoreADCS):
-                while len(buffer) > 0 and len(squishedpackets) < config['telemetry']['max_packet_size']:
-                    # test = buffer.pop()
-                    # print(test)
-                    # squishedPackets += test
-                    squishedpackets += buffer.pop()
-
-            # TODO: alternate between radios
-            logger.debug(squishedpackets)
-            radio_output.send(squishedpackets, radio)
-            squishedpackets = ""
-            time.sleep(6)
-
-    # packet_lock.release()
-
-
-@command("telem_clear")
-def clear_buffers():
-    global packet_lock, event_packet_buffer, telem_packet_buffer
-    with packet_lock:
-        event_packet_buffer.clear()
-        telem_packet_buffer.clear()
+    global packet_lock
+    while True:
+        with packet_lock:
+            message = general_queue.popleft()
+            if type(message) is str and message[0] == ';':
+                command_ingest.enqueue(message[4:])
+            elif type(message) is error.Error:
+                err_stack.append(message)
+            elif type(message) is log.Log:
+                log_stack.append(message)
+            else:   # Shouldn't execute (enqueue() should catch it) but here just in case
+                logger.error("Message prefix invalid.")
 
 
 def start():
     """
     Starts the telemetry send thread
     """
-    t2 = ThreadHandler(target=partial(telemetry_send),
-                       name="telemetry-telemetry_send")
-    t2.start()
+    threadDecide = ThreadHandler(target=partial(decide()),
```

Comment:
The `partial()` function takes a function object, not a function call. 

Recommended Change:
`threadDecide = ThreadHandler(target=partial(decide), name="telemetry-decide")`

Reply:
Done in 963480f